client.go

```diff
@@ -250,74 +249,65 @@ func (np *PubNameAndPassword) PubKey() (e4crypto.Ed25519PublicKey, error) {
 // depending the given ClientConfig
 //
 // config is a ClientConfig, either SymIDAndKey, SymNameAndPassword, PubIDAndKey or PubNameAndPassword
-// persistStatePath is the file system path to the file to read and persist the client's state.
-func NewClient(config ClientConfig, persistStatePath string) (Client, error) {
-	return config.genNewClient(persistStatePath)
+// store is an e4.ReadWriteSeeker implementation
+func NewClient(config ClientConfig, store ReadWriteSeeker) (Client, error) {
+	return config.genNewClient(store)
 }
 
 // newClient creates a new client, generating a random ID if they are empty
-func newClient(id []byte, clientKey keys.KeyMaterial, persistStatePath string) (Client, error) {
+func newClient(id []byte, clientKey keys.KeyMaterial, store ReadWriteSeeker) (Client, error) {
 	if len(id) == 0 {
 		return nil, errors.New("client id must not be empty")
 	}
 
 	c := &client{
 		Key:            clientKey,
 		TopicKeys:      make(map[string]keys.TopicKey),
-		FilePath:       persistStatePath,
 		ReceivingTopic: TopicForID(id),
+
+		store: store,
 	}
 
 	c.ID = make([]byte, len(id))
 	copy(c.ID, id)
 
 	log.SetPrefix("e4client\t")
 
+	if err := c.save(); err != nil {
+		return nil, err
+	}
+
 	return c, nil
 }
 
 // LoadClient loads a client state from the file system
-func LoadClient(persistStatePath string) (Client, error) {
+func LoadClient(store ReadWriteSeeker) (Client, error) {
 	c := &client{}
-	err := readJSON(persistStatePath, c)
+
+	decoder := json.NewDecoder(store)
+	err := decoder.Decode(c)
 	if err != nil {
 		return nil, err
 	}
 
+	store.Seek(0, io.SeekStart)
```

Comment:
Let’s not ignore errors from Seek since that’ll determine whether the next store use will be valid or not:
  if _, err := store.Seek(0, io.SeekStart)